lib/utils/exceptions.dart

```diff
@@ -0,0 +1,45 @@
+String extractExceptionMessage(Object exception) {
+  if (exception is String) {
+    final message = _extractMessageFromString(exception);
+    if (message != null) {
+      return message;
+    }
+  }
+
+  return exception.toString();
+}
+
+String _extractMessageFromString(String exception) {
+  for (final helper in [
+    _ExtractMessageFromStringRegexHelper(r'{\"detail\":\".*\"}', r'{\"detail\":\"', r'\"}'),
+    _ExtractMessageFromStringRegexHelper(r'{NSLocalizedDescription=.*}', r'{NSLocalizedDescription=', r'}'),
+    _ExtractMessageFromStringRegexHelper(r'LNURL: .* method:', r'LNURL: ', r' method:'),
+  ]) {
+    final message = helper.extract(exception);
+    if (message != null) {
+      return message;
+    }
+  }
+
+  return null;
+}
```

Comment:
I believe we can use a single regex to extract the message instead of matching through a loop of _ExtractMessageFromStringRegexHelper's and calling extract function to remove lhs & rhs with replace operations.

Something like:
`((?<={NSLocalizedDescription=)(.*)(?=}))|((?<=LNURL:.)(.*)(?=.method:))|((?<=\"detail\":\")(.*)(?=.*\"}))`

It uses capture groups to combine the regexp's that we're checking against and extracts the message with the use of lookaround assertions.

We can define group strings on their own and join them with "|" for readability.

Reply:
Grouping like that, how can we remove the leading and trailing parts of the string?

Right now I'm doing that:
`match.group(0).replaceAll(RegExp(lhsReplace), '').replaceAll(RegExp(rhsReplace), '');`

Reply:
The regex I mentioned make use of lookaround assertions(look behind, look ahead) to retrieve the part that's between leading and trailing characters. It returns the match only if it's preceeded and followed by those assertions.
Here's a straightforward source on lookaround assertions: https://javascript.info/regexp-lookahead-lookbehind

Reply:
Interesting, I believe I managed to apply it; not sure if there is a better way.

Reply:
I believe we can merge extractExceptionMessage and _extractMessageFromString into a single function. There's also no need to go through all matches, using grouped.stringMatch(exception) directly to get first match(or null) is sufficient.

Reply:
Ahh, much better.

About the methods, the idea is to have different extract strategies given the error type; I did apply your suggestion for now as we have only String; next time I add a new style, I can break it again; there is no need to overengineering for now.